### PR TITLE
Feature: Auto update feature of FilesLauncher

### LINKED ISSUE #11746: Feature: Keep FilesLauncher.exe up to date on users devices
### What feature or improvement do you think would benefit Files?

Whenever we make changes to the FilesLauncher file, existing users need to toggle the default file manager setting off and on for it to apply. Automatically updating this file will make sure users always have the latest improvements.

### Requirements

- Keep the FilesLauncher.exe up to date when users update Files

### Files Version

v2.4.40

### Windows Version

Windows 11

### Comments

_No response_

### PR DESCRIPTION
**Pull request by hishitetsu**
**Resolved / Related Issues**
- [X] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #11746 

**Validation**
How did you test these changes?
- [X] Did you build the app and test your changes?
- [X] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [X] Are there any other steps that were used to validate these changes?
   You can try this feature by editing or deleting FilesLaucher.exe.sha256 in the LocalAppData folder.
   A log is output when updated.

**Details**
* The update process runs when the app is launched with the default file manager setting turned on.
* This feature checks the file stored the SHA256 hash value of FilesLauncher.exe, rather than FilesLauncher.exe itself.
* FilesLaucher.exe.sha256 is generated during the build process of FilesLauncher.

### CODE REVIEW

File: src/Files.App/App.xaml.cs [hunk 1]
```diff
@@ -226,6 +226,7 @@ private static async Task InitializeAppComponentsAsync()
 			var updateService = Ioc.Default.GetRequiredService<IUpdateService>();
 			await updateService.CheckForUpdates();
 			await updateService.DownloadMandatoryUpdates();
+			await updateService.CheckAndUpdateFilesLauncherAsync();
```
Comment (yaira2) [inline src/Files.App/App.xaml.cs:229]: Does this run after startup is complete?

Comment (hishitetsu) [reply src/Files.App/App.xaml.cs:229]: Called asynchronously during startup.

Comment (yaira2) [reply src/Files.App/App.xaml.cs:229]: Can we delay it until after startup? It probably doesn't take long but we might as well delay it.

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**